test/plugin_test.dart

```diff
@@ -14,50 +16,119 @@ void main() {
   late MockProcessManager pm;
   late FileSystem fs;
 
+  const cli = MockCLI.name;
+  const orgAndProject = '--org o --project p';
+  const project = 'project';
+  const version = '1.1.0';
+  const release = '$project@$version';
+  const buildDir = '/subdir';
+
   setUp(() {
     // override dependencies for testing
     pm = MockProcessManager();
     injector.registerSingleton<ProcessManager>(() => pm, override: true);
     fs = MemoryFileSystem.test();
+    fs.currentDirectory = fs.directory(buildDir)..createSync();
     injector.registerSingleton<FileSystem>(() => fs, override: true);
+    injector.registerSingleton<CLISetup>(() => MockCLI(), override: true);
   });
 
-  test('fails without args and pubspec', () async {
-    final exitCode = await plugin.run([]);
-    expect(exitCode, 1);
-    expect(pm.commandLog, const [
-      'chmod +x .dart_tool/pub/bin/sentry_dart_plugin/sentry-cli',
-      '.dart_tool/pub/bin/sentry_dart_plugin/sentry-cli help'
-    ]);
-  });
+  Future<Iterable<String>> runWith(String config) async {
+    // properly indent the configuration for the `sentry` section in the yaml
+    final configIndented =
+        config.trim().split('\n').map((l) => '  ${l.trim()}').join('\n');
 
-  test('works with pubspec', () async {
     fs.file('pubspec.yaml').writeAsStringSync('''
-name: project
-version: 1.1.0
+name: $project
+version: $version
 
 sentry:
-  upload_native_symbols: true
-  include_native_sources: true
-  upload_source_maps: true
-  auth_token: t
+  auth_token: t # TODO: support not specifying this, let sentry-cli use the value it can find in its configs
```

Comment:
The problem was that this plugin could not log a proper error, sentry CLI outputs too much stuff when something goes wrong.

Reply:
That can be solved too. I just didn't want to do this unrelated thing in the same PR.

Also relates to https://github.com/getsentry/sentry-dart-plugin/issues/23

Reply:
Yes, but I didn't want to during the hack week, we can leave this open and fix it later.
My message was more about giving context on why it's like this.